tests/garage/tf/models/test_new_model.py

```diff
@@ -0,0 +1,77 @@
+import pickle
+
+import numpy as np
+import tensorflow as tf
+
+from garage.tf.models.gaussian_mlp_model import GaussianMLPModel
+from tests.fixtures import TfGraphTestCase
+
+
+class TestNewModel(TfGraphTestCase):
+    def test_model_creation(self):
+        input_var = tf.placeholder(tf.float32, shape=(None, 5))
+        model = GaussianMLPModel(output_dim=2)
+        model.build(input_var)
+        data = np.random.random((3, 5))
+        self.sess.run(model.output, feed_dict={model.input: data})
+
+    def test_model_creation_with_custom_name(self):
+        input_var = tf.placeholder(tf.float32, shape=(None, 5))
+        model = GaussianMLPModel(output_dim=2)
+        model.build(input_var, name="custom_model")
```

Comment:
it doesn't seem to me that you ever interact with the outputs from `build` in these tests.

you also never test a hierarchical model (a model with a model as input to build), which i think is a key use case.